runtime/lua/vim/lsp/util.lua

```diff
@@ -1036,17 +1045,18 @@ function M.make_floating_popup_options(width, height, opts)
   }
 end
 
---- Jumps to a location.
+--- Shows document and optionally jumps to the location.
 ---
 ---@param location table (`Location`|`LocationLink`)
----@param offset_encoding string utf-8|utf-16|utf-32 (required)
----@param reuse_win boolean Jump to existing window if buffer is already opened.
----@returns `true` if the jump succeeded
-function M.jump_to_location(location, offset_encoding, reuse_win)
+---@param offset_encoding "utf-8" | "utf-16" | "utf-32"
+---@param reuse_win boolean Jump to existing window if buffer is already open.
+---@param focus boolean | nil Whether to focus/jump to location if possible. Defaults to true.
+---@return boolean `true` if succeeded
+function M.show_document(location, offset_encoding, reuse_win, focus)
```

Comment:
I can imagine that people will want to customize how the new window gets opened. There's no need to do that in this PR but maybe `offset_encoding`, `reuse_win` and `focus` could be put into an `opts` table for easier extension later on. What do you think?

Reply:
Makes sense. Perhaps `function M.show_document(location, offset_encoding, opts)`, since `offset_encoding` is required?